pyproject.toml

```diff
@@ -49,6 +50,12 @@ select = [
     "UP039", # don't use parens after a class definition with no bases
 ]
 
+[tool.ruff.lint.extend-per-file-ignores]
```

Comment:
I'd prefer if we just ignored all flake8-bugbear rules on .pyi files, like we do currently. We know none of them are really relevant to stubs. Ideally, ruff wouldn't emit any false positives on stub files if any of these rules are enabled globally, but that's bug-prone, and I don't really see a good reason for us to keep any of these enabled for .pyi files. (Unless you have some specifically in mind that you think would be useful on stubs? :)

Reply:
Since Ruff does often care about differentiating `.py` from `.pyi` and special cases some rules, I was also trying to see if we could simplify the config and just turn it on w/o exception

As far as rules that can apply to stubs, I found a handful, but nothing that applies to typeshed, except maybe B0033

There's B018 [useless-expression](https://docs.astral.sh/ruff/rules/useless-expression/) . But it's already handled by pyright [reportInvalidStubStatement](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportInvalidStubStatement) & [reportUnusedExpression](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportUnusedExpression) both trigger

B021 [f-string-docstring](https://docs.astral.sh/ruff/rules/f-string-docstring/) but Typeshed doesn't include docstrings

The following can be valuable in stubs to catch actual issues in source, but there's nothing typeshed can do about it (so doesn't concern us as a third-party stubs repo)
B006 [mutable-argument-default](https://docs.astral.sh/ruff/rules/mutable-argument-default/)
B019 [cached-instance-method](https://docs.astral.sh/ruff/rules/cached-instance-method/)
B024 [abstract-base-class-without-abstract-method](https://docs.astral.sh/ruff/rules/abstract-base-class-without-abstract-method/)

B033 [duplicate-value](https://docs.astral.sh/ruff/rules/duplicate-value/) is technically not caught by any other rule. But also is a useless statement in stubs.
```py
TEST: Final = {1, 2, 3, 1} # kindof useless anyway since the type is `set[int]`, could be caught by flake8-PYI

def foo(bar: set[int] = {1, 2, 3, 1}): ... # Actually kindof somewhat useful for default values? Ignoring B006 this currently passes checks.
```

Reply:
Alright, I guess I'd be open to enabling B033! That seems useful. But I'd still prefer to keep the others disabled on our `.pyi` files for now; we'll have less chance of running into bugs like https://github.com/astral-sh/ruff/issues/10083

Reply:
> people writing rules primarily for .py files won't necessarily remember that .pyi files also exist :)

Yeah I had to raise a few of those issues ^^

Unless you know of a clean way to only enable `B033` for pyi, I'll just drop that one too.